lib/iris/aux_factory.py

```diff
@@ -817,30 +822,21 @@ def make_coord(self, coord_dims_func):
 
         # Build the points array.
         nd_points_by_key = self._remap(dependency_dims, derived_dims)
-        points_shape = self._shape(nd_points_by_key)
-
-        # Calculate the nsigma slice.
-        nsigma_slice = [slice(None)] * len(derived_dims)
-        dim, = dependency_dims['zlev']
-        index = derived_dims.index(dim)
-        nsigma_slice[index] = slice(0, int(nd_points_by_key['nsigma']))
-        nsigma_slice = tuple(nsigma_slice)
 
+        nsigma, = nd_points_by_key['nsigma']
         points = self._derive(nd_points_by_key['sigma'],
                               nd_points_by_key['eta'],
                               nd_points_by_key['depth'],
                               nd_points_by_key['depth_c'],
                               nd_points_by_key['zlev'],
-                              points_shape,
-                              nsigma_slice)
+                              nsigma,
```

Comment:
Because I wanted it to be seen as an "extra" argument, i.e. a normal Python value, and not another dependency array.